### DISCUSSION TITLE
Grid columns do not grow/shrink with grid state feature

### DISCUSSION THREAD

**Discussion by palrajarockiaraj** [IgniteUI/igniteui-angular#14678] [Q&A] [answered]
The grid columns do not grow/shrink when the state of the grid is saved and the size of the grid has been changed manually using a splitter component.

**Comment by ddaribo** [comment]
Hello @palrajarockiaraj,

In an effort to reproduce the described scenario, I created [this sample](https://stackblitz.com/edit/igx-grid-state-splitter?file=src%2Fapp%2Fgrid%2Fgrid-save-state%2Fgrid-state.component.html) demonstrating an `IgxGrid` nested within a horizontal `IgxSplitterPane` in addition to the state-persistence feature.
Moreover, the columns widths are set to percentage values, such that the initial number of columns take up the entire available space in the grid. As you will see, they do grow and shrink when resizing the grid with the splitter (as its width is set to 100% of the parent container). Additionally, this behavior is also present after saving and restoring the grid state:

![discussion-example](https://github.com/user-attachments/assets/fc140ed2-66b3-482f-ab09-a17a205e57f8)

Hence, I am wondering what are the columns sizing settings in your app in this case? Is it possible that when restoring the state, their width settings get overridden to something else?
Regarding column sizing and the available options, I can suggest reviewing [this section](https://www.infragistics.com/products/ignite-ui-angular/angular/components/grid/sizing#column-sizing) of the IgxGrid Sizing documentation. 

In conclusion, please, check out the referenced resources and let me know if you still need assistance on the matter. If so, please, provide more information about your configuration. Ideally, you could fork and modify the above referenced sample with only the relevant code in order to reproduce the issue. Having a working sample on our side, which we can debug will be highly appreciated so that we can proceed with assisting you.

**Reply by palrajarockiaraj** [reply]
The issue happens when the columns' widths are set in pixels instead of percentages.

I created [this sample](https://stackblitz.com/edit/igx-grid-state-splitter-nue7lj) by forking and modifying it as per my requirement.

The columns do not grow to occupy the available width when the grid's width grows.

**Comment by ddaribo** [comment, marked as answer]
Hi @palrajarockiaraj,

Thank you for following up!

According to the previously referenced [Column sizing section](https://www.infragistics.com/products/ignite-ui-angular/angular/components/grid/sizing#column-sizing) in the Grid Sizing documentation topic, when a [pixel width](https://www.infragistics.com/products/ignite-ui-angular/angular/components/grid/sizing#pixels-2) is set to columns: 

_"they stick to that size, unless they are resized manually. Since the combined width of the columns is static, it can be less than the IgxGrid width or exceed it."_

, which means that what you are observing is expected and by design.

If you like to achieve a behavior, where the columns grow and shrink along with the grid width, setting [percentage values](https://www.infragistics.com/products/ignite-ui-angular/angular/components/grid/sizing#percentages-2) is the approach to adopt.
